### PR TITLE
Cleanup and more docs

### PR DESCRIPTION
**Pull request by gabriel**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,6 +113,19 @@ YOLayout is a great fit for custom drawn controls. Views that render in `drawRec
...
 Nope. If your layout is really simple, or it doesn't have dynamic sizing, just use `layoutSubviews` (UIKit) or `layout` (AppKit) like you normally would. YOLayout doesn't do anything special to override behavior and is compatible with existing layout methods.
 
+## Disadvantages of using YOLayout
+
+YOLayout, like most things, has trade-offs. We like using it especially for really complex layouts with lots of different alignments and for things that are dynamically sized. 
+
+But there are downsides:
+
+- You might find yourself hardcoding pixel values for things like padding, a lot.
+- If you are tweaking padding and positioning, you have to re-run the project to visualize those changes, whereas in Interface Builder you would see those changes immediately.
```
Comment (johnboiles) [inline README.md]: Not necessarily if you make your view an `IB_DESIGNABLE` and all your subviews are drawable! I've been doing this a lot lately so I can test views without a full compile. You can make a throwaway xib and instantiate your view there to get a preview rendering.

That said, that's pretty nuanced, so we should probably still keep this as a disadvantage.

File: README.md [hunk 2]
```diff
@@ -109,6 +113,19 @@ YOLayout is a great fit for custom drawn controls. Views that render in `drawRec
...
+- Autolayout works really well in certain cases and easier to read than manual layout code.
+- YOLayout is a custom framework whereas Interface Builder and AutoLayout are part of Apple and their SDK.
+
+And probably many more, feel free to tell us why you hate it, by submitting an [issue](https://github.com/YOLayout/YOLayout/issues).
```
Comment (johnboiles) [inline README.md]: Humility++

File: YOLayout/YOLayout_IMP.h [hunk 3]
```diff
@@ -148,44 +138,17 @@ typedef enum {
 - (CGRect)setSize:(CGSize)size inRect:(CGRect)inRect view:(id)view options:(YOLayoutOptions)options;
 
 /*!
- Set origin.
+ Set origin. Same as setFrame:view: but uses the existing view.frame.size.
+
+ @param origin Origin
+ @param view View
  */
 - (CGRect)setOrigin:(CGPoint)origin view:(id)view;
```
Comment (johnboiles) [inline YOLayout/YOLayout_IMP.h]: Do you use this at all? I think I'd be ok with removing it. Or making it `setOrigin:inRect:view:options:` to make it more generally useful

File: YOLayout/YOLayout_IMP.h [hunk 4]
```diff
@@ -148,44 +138,17 @@ typedef enum {
...
-- (CGRect)setOrigin:(CGPoint)origin view:(id)view sizeToFit:(BOOL)sizeToFit;
 
 /*!
- Set size.
+ Set size. Same as setFrame:view: but uses the existing view.frame.origin.
  */
 - (CGRect)setSize:(CGSize)size view:(id)view;
```
Comment (johnboiles) [inline YOLayout/YOLayout_IMP.h]: I don't think I use this one either.

**Comment by johnboiles** [comment]
Thanks for doing this!!

File: README.md (hunk 1 above)
Comment (gabriel) [inline README.md]: oh awesome... I'll throw in an "except if its IB_DESIGNABLE" :)

File: YOLayout/YOLayout_IMP.h (hunk 3 above)
Comment (gabriel) [inline YOLayout/YOLayout_IMP.h]: oh yeah I see how options might be useful, will use that one

**Pull request merged**